server/channels/store/localcachelayer/user_layer.go

```diff
@@ -85,6 +97,46 @@ func (s *LocalCacheUserStore) InvalidateProfilesInChannelCache(channelID string)
 	}
 }
 
+func (s *LocalCacheUserStore) GetAllProfiles(options *model.UserGetOptions) ([]*model.User, error) {
+	// We check to see if any of the options are set or not, and then
+	// use the cache only if none are set, which is the most common case.
+	// options.WithoutTeam, Sort is unused
+	if options.InTeamId == "" &&
+		options.NotInTeamId == "" &&
+		options.InChannelId == "" &&
+		options.NotInChannelId == "" &&
+		options.InGroupId == "" &&
+		options.NotInGroupId == "" &&
+		!options.GroupConstrained &&
+		!options.Inactive &&
+		!options.Active &&
+		options.Role == "" &&
+		len(options.Roles) == 0 &&
+		len(options.ChannelRoles) == 0 &&
+		len(options.TeamRoles) == 0 &&
+		options.Page == 0 && options.PerPage == 100 && // This is hardcoded to the webapp call.
+		options.ViewRestrictions == nil {
+		// read from cache
+		var users []*model.User
+		if err := s.rootStore.doStandardReadCache(s.rootStore.allUserCache, allUserKey, &users); err == nil {
```

Comment:
Does this cache invalidated whenever a new user created?

Reply:
It does. app/user.go:267, but only if the Email is verified. I'll fix it.